node/flatpak_node_generator/providers/npm.py

```diff
@@ -369,9 +367,49 @@ async def generate_package(self, package: Package) -> None:
             # Get a unique name to use for the Git repository folder.
             name = f'{package.name}-{source.commit}'
             path = self.gen.data_root / 'git-packages' / name
-            self.git_sources[package.lockfile][path] = source
+            self.git_sources[package.lockfile][path] = (package.name, source)
             self.gen.add_git_source(source.url, source.commit, path)
 
+            url = urllib.parse.urlparse(source.url)
+            if url.netloc == 'git@github.com' or url.netloc == 'github.com':
```

Comment:
could be `url.hostname == 'github.com'` (and maybe also check `url.username in ('git', '')` or similar? do you know if that's semantically important in any way? if not, then just checking the hostname is a bit cleaner.) ditto for the `gitlab.com` check below.